packages/web/src/components/home/token-info/TokenInfo.tsx

```diff
@@ -63,38 +63,77 @@ const TokenInfo: React.FC<TokenInfoProps> = ({ item, idx }) => {
     mostLiquidPool,
     last7days,
     graphStatus,
+    isNative,
   } = item;
   const router = useRouter();
+  const theme = useTheme();
 
   const onClickItem = (path: string) => {
     router.push("/tokens/" + makeId(path));
   };
 
   const onClickPoolItem = (item: MostLiquidPool) => {
-    const poolPath = `${item.tokenPair.tokenA.path}:${
-      item.tokenPair.tokenB.path
-    }:${Number(item.feeRate.slice(0, item.feeRate.length - 1)) * 10000}`;
+    const poolPath = `${item.tokenPair.tokenA.path}:${item.tokenPair.tokenB.path
+      }:${Number(item.feeRate.slice(0, item.feeRate.length - 1)) * 10000}`;
     if (item.tokenPair.tokenA.logoURI) {
       location.href = `/earn/pool/${makeId(poolPath)}`;
     }
   };
 
+  const tokenPathDisplay = useMemo(() => {
+    const path_ = path;
+
+    if (isNative) return "Native coin";
+
+    const tokenPathArr = path_?.split("/") ?? [];
+
+    if (tokenPathArr?.length <= 0) return path_;
+
+    const lastPath = tokenPathArr[tokenPathArr?.length - 1];
+
+    if (lastPath.length >= 12) {
+      return "..." + tokenPathArr[tokenPathArr?.length - 1].slice(length - 12, length - 1);
+    }
+
+    return path_.replace("gno.land", "...");
+  }, [isNative, path]);
+
+  const onClickPath = useCallback((e: React.MouseEvent<HTMLDivElement, MouseEvent>, path: string) => {
+    e.stopPropagation();
+    if (path === "gnot") {
+      window.open("https://gnoscan.io/", "_blank");
+    } else {
+      window.open("https://gnoscan.io/tokens/" + makeId(path), "_blank");
```

Comment:
maybe `makeId(path), "blank");` ?

The URL in GnoScan is like this: https://gnoscan.io/tokens/gno.land/r/demo/tong

Reply:
Or maybe the "_" is not related, but the current PR sends is to `https://gnoscan.io/tokens/gno.land_r_demo_baz`, which is incorrect. `-` should be `/`.